src/DateInput/DateInput.jsx

```diff
@@ -0,0 +1,75 @@
+import React from "react";
+import classnames from "classnames";
+import moment from "moment";
+
+import "./DateInput.less";
+
+import DatePicker from "../DatePicker/DatePicker";
+import {Button} from "../Button/Button";
+import {FlexBox, FlexItem} from "../flex";
+
+export class DateInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showPicker: false,
+    };
+  }
+
+  _onChange(val) {
+    this.props.onChange(val);
+    this.setState({
+      showPicker: false,
+    });
+  }
+
+  render() {
+    const val = this.props.value ? moment(this.props.value) : moment();
+
+    return (<FlexBox column className={classnames("DateInput", this.props.className)}>
+      <div className="DateInput--InfoRow">{this.props.label}</div>
+      <FlexBox className="DateInput--DateRow">
+        <span className="DateInput--Date">
+          {this.props.value ? val.format("M/D/YYYY") : "--"}
```

Comment:
Don't use `--` as an empty state. Probably will be fixed when you convert to using the correct input type.